src/console_op/data.go

```diff
@@ -34,13 +34,34 @@ import (
 	"log"
 	"net/http"
 	"time"
+
+	"github.com/go-chi/chi/v5"
 )
 
 // Variable to hold address of console-data service
 var dataAddrBase string = "http://cray-console-data/v1"
 
+type DataService interface {
+	dataAddNodes(newNodes []nodeConsoleInfo) bool
+	dataRemoveNodes(removedNodes []nodeConsoleInfo)
+	checkHeartbeats()
+	doGetPodLocation(w http.ResponseWriter, r *http.Request)
+	doGetNodePod(w http.ResponseWriter, r *http.Request)
+	getNodePodForXname(xname string) (string, error)
+}
+
+// Implements DataService
+type DataManager struct {
+	k8Service K8Service
+}
+
+// Constructor injection for dependencies
+func NewDataManager(k8s K8Service) *DataManager {
+	return &DataManager{k8Service: k8s}
+}
+
 // function to interact with console-data api to add new nodes to the db
-func dataAddNodes(newNodes []nodeConsoleInfo) bool {
+func (DataManager) dataAddNodes(newNodes []nodeConsoleInfo) bool {
```

Comment:
Why makes these DataManager functions?  It looks like this is just for readability, grouping similar functions, but doesn't impact functionality, is that right?

Reply:
That is correct it should not affect any functionality, its for easier dependency reading and as well as the ability to mock out the calls in testing since we can override behaviors in each service/manager.